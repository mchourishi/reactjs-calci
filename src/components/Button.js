import React from "react";
import PropTypes from "prop-types";

class Button extends React.Component {
  handleClick = () => {
    this.props.clickHandler(this.props.name);
  };
  render() {
    const className = ["calculator-key", this.props.cls ? this.props.cls : ""];

    return (
      <div>
        <button onClick= {this.handleClick} className={className.join(" ").trim()}>
          {this.props.name}
        </button>
      </div>
    );
  }
}

Button.propTypes = {
  name: PropTypes.string,
  cls: PropTypes.string,
  clickHandler: PropTypes.func
};
export default Button;
