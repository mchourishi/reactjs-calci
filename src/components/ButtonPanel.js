import React from "react";
import Button from "./Button";
import PropTypes from "prop-types";

class ButtonPanel extends React.Component {
  handleClick = buttonName => {
    this.props.clickHandler(buttonName);
  };
  render() {
    return (
      <div className="calculator-keypad">
        <div className="input-keys">
          <div className="function-keys">
            <Button name="AC" clickHandler={this.handleClick} />
            <Button name="⌫" clickHandler={this.handleClick} />
            <Button name="+/-" clickHandler={this.handleClick} />
            <Button name="√" clickHandler={this.handleClick} />
            <Button name="x²" clickHandler={this.handleClick} />
            <Button name="%" clickHandler={this.handleClick} />
          </div>
          <div className="digit-keys">
            <Button name="0" cls="key-0" clickHandler={this.handleClick} />
            <Button name="." clickHandler={this.handleClick} />
            <Button name="1" clickHandler={this.handleClick} />
            <Button name="2" clickHandler={this.handleClick} />
            <Button name="3" clickHandler={this.handleClick} />
            <Button name="4" clickHandler={this.handleClick} />
            <Button name="5" clickHandler={this.handleClick} />
            <Button name="6" clickHandler={this.handleClick} />
            <Button name="7" clickHandler={this.handleClick} />
            <Button name="8" clickHandler={this.handleClick} />
            <Button name="9" clickHandler={this.handleClick} />
          </div>
        </div>
        <div className="operator-keys">
          <Button name="/" clickHandler={this.handleClick} />
          <Button name="*" clickHandler={this.handleClick} />
          <Button name="-" clickHandler={this.handleClick} />
          <Button name="+" clickHandler={this.handleClick} />
          <Button name="=" cls="equalkey" clickHandler={this.handleClick} />
        </div>
      </div>
    );
  }
}

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func
};

export default ButtonPanel;
