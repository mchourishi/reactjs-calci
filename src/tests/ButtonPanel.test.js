import React from "react";
import { shallow } from "enzyme";
import ButtonPanel from "../components/ButtonPanel";
import Button from "../components/Button";

describe("Button Panel", () => {
  let wrapper;
  beforeEach(() => (wrapper = shallow(<ButtonPanel clickHandler={jest.fn()} />)));

  it("should render correctly", () => expect(wrapper).toMatchSnapshot());

  it("should render the Button Component", () => {
    expect(wrapper.containsMatchingElement(<Button />)).toEqual(true);
  });

  it("should render total 22 Buttons", () => {
    expect(wrapper.find(Button).length).toEqual(22);
  });

  it("should render 6 function keys",()=>{
    expect(wrapper.find('.function-keys').children().length).toEqual(6);
  });

  it("should render 11 digit keys",()=>{
    expect(wrapper.find('.digit-keys').children().length).toEqual(11);
  });

  it("should render 5 operator keys",()=>{
    expect(wrapper.find('.operator-keys').children().length).toEqual(5);
  });
});
