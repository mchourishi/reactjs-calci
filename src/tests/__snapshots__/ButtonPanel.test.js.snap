// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Panel should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonPanel
    clickHandler={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="input-keys"
        >
          <div
            className="function-keys"
          >
            <Button
              clickHandler={[Function]}
              name="AC"
            />
            <Button
              clickHandler={[Function]}
              name="⌫"
            />
            <Button
              clickHandler={[Function]}
              name="+/-"
            />
            <Button
              clickHandler={[Function]}
              name="√"
            />
            <Button
              clickHandler={[Function]}
              name="x²"
            />
            <Button
              clickHandler={[Function]}
              name="%"
            />
          </div>
          <div
            className="digit-keys"
          >
            <Button
              clickHandler={[Function]}
              cls="key-0"
              name="0"
            />
            <Button
              clickHandler={[Function]}
              name="."
            />
            <Button
              clickHandler={[Function]}
              name="1"
            />
            <Button
              clickHandler={[Function]}
              name="2"
            />
            <Button
              clickHandler={[Function]}
              name="3"
            />
            <Button
              clickHandler={[Function]}
              name="4"
            />
            <Button
              clickHandler={[Function]}
              name="5"
            />
            <Button
              clickHandler={[Function]}
              name="6"
            />
            <Button
              clickHandler={[Function]}
              name="7"
            />
            <Button
              clickHandler={[Function]}
              name="8"
            />
            <Button
              clickHandler={[Function]}
              name="9"
            />
          </div>
        </div>,
        <div
          className="operator-keys"
        >
          <Button
            clickHandler={[Function]}
            name="/"
          />
          <Button
            clickHandler={[Function]}
            name="*"
          />
          <Button
            clickHandler={[Function]}
            name="-"
          />
          <Button
            clickHandler={[Function]}
            name="+"
          />
          <Button
            clickHandler={[Function]}
            cls="equalkey"
            name="="
          />
        </div>,
      ],
      "className": "calculator-keypad",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="function-keys"
            >
              <Button
                clickHandler={[Function]}
                name="AC"
              />
              <Button
                clickHandler={[Function]}
                name="⌫"
              />
              <Button
                clickHandler={[Function]}
                name="+/-"
              />
              <Button
                clickHandler={[Function]}
                name="√"
              />
              <Button
                clickHandler={[Function]}
                name="x²"
              />
              <Button
                clickHandler={[Function]}
                name="%"
              />
            </div>,
            <div
              className="digit-keys"
            >
              <Button
                clickHandler={[Function]}
                cls="key-0"
                name="0"
              />
              <Button
                clickHandler={[Function]}
                name="."
              />
              <Button
                clickHandler={[Function]}
                name="1"
              />
              <Button
                clickHandler={[Function]}
                name="2"
              />
              <Button
                clickHandler={[Function]}
                name="3"
              />
              <Button
                clickHandler={[Function]}
                name="4"
              />
              <Button
                clickHandler={[Function]}
                name="5"
              />
              <Button
                clickHandler={[Function]}
                name="6"
              />
              <Button
                clickHandler={[Function]}
                name="7"
              />
              <Button
                clickHandler={[Function]}
                name="8"
              />
              <Button
                clickHandler={[Function]}
                name="9"
              />
            </div>,
          ],
          "className": "input-keys",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Button
                  clickHandler={[Function]}
                  name="AC"
                />,
                <Button
                  clickHandler={[Function]}
                  name="⌫"
                />,
                <Button
                  clickHandler={[Function]}
                  name="+/-"
                />,
                <Button
                  clickHandler={[Function]}
                  name="√"
                />,
                <Button
                  clickHandler={[Function]}
                  name="x²"
                />,
                <Button
                  clickHandler={[Function]}
                  name="%"
                />,
              ],
              "className": "function-keys",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": "AC",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": "⌫",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": "+/-",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": "√",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": "x²",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": "%",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Button
                  clickHandler={[Function]}
                  cls="key-0"
                  name="0"
                />,
                <Button
                  clickHandler={[Function]}
                  name="."
                />,
                <Button
                  clickHandler={[Function]}
                  name="1"
                />,
                <Button
                  clickHandler={[Function]}
                  name="2"
                />,
                <Button
                  clickHandler={[Function]}
                  name="3"
                />,
                <Button
                  clickHandler={[Function]}
                  name="4"
                />,
                <Button
                  clickHandler={[Function]}
                  name="5"
                />,
                <Button
                  clickHandler={[Function]}
                  name="6"
                />,
                <Button
                  clickHandler={[Function]}
                  name="7"
                />,
                <Button
                  clickHandler={[Function]}
                  name="8"
                />,
                <Button
                  clickHandler={[Function]}
                  name="9"
                />,
              ],
              "className": "digit-keys",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "cls": "key-0",
                  "name": "0",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": ".",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": "1",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": "2",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": "3",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": "4",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": "5",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": "6",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": "7",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": "8",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "clickHandler": [Function],
                  "name": "9",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              clickHandler={[Function]}
              name="/"
            />,
            <Button
              clickHandler={[Function]}
              name="*"
            />,
            <Button
              clickHandler={[Function]}
              name="-"
            />,
            <Button
              clickHandler={[Function]}
              name="+"
            />,
            <Button
              clickHandler={[Function]}
              cls="equalkey"
              name="="
            />,
          ],
          "className": "operator-keys",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "clickHandler": [Function],
              "name": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "clickHandler": [Function],
              "name": "*",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "clickHandler": [Function],
              "name": "-",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "clickHandler": [Function],
              "name": "+",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "clickHandler": [Function],
              "cls": "equalkey",
              "name": "=",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="input-keys"
          >
            <div
              className="function-keys"
            >
              <Button
                clickHandler={[Function]}
                name="AC"
              />
              <Button
                clickHandler={[Function]}
                name="⌫"
              />
              <Button
                clickHandler={[Function]}
                name="+/-"
              />
              <Button
                clickHandler={[Function]}
                name="√"
              />
              <Button
                clickHandler={[Function]}
                name="x²"
              />
              <Button
                clickHandler={[Function]}
                name="%"
              />
            </div>
            <div
              className="digit-keys"
            >
              <Button
                clickHandler={[Function]}
                cls="key-0"
                name="0"
              />
              <Button
                clickHandler={[Function]}
                name="."
              />
              <Button
                clickHandler={[Function]}
                name="1"
              />
              <Button
                clickHandler={[Function]}
                name="2"
              />
              <Button
                clickHandler={[Function]}
                name="3"
              />
              <Button
                clickHandler={[Function]}
                name="4"
              />
              <Button
                clickHandler={[Function]}
                name="5"
              />
              <Button
                clickHandler={[Function]}
                name="6"
              />
              <Button
                clickHandler={[Function]}
                name="7"
              />
              <Button
                clickHandler={[Function]}
                name="8"
              />
              <Button
                clickHandler={[Function]}
                name="9"
              />
            </div>
          </div>,
          <div
            className="operator-keys"
          >
            <Button
              clickHandler={[Function]}
              name="/"
            />
            <Button
              clickHandler={[Function]}
              name="*"
            />
            <Button
              clickHandler={[Function]}
              name="-"
            />
            <Button
              clickHandler={[Function]}
              name="+"
            />
            <Button
              clickHandler={[Function]}
              cls="equalkey"
              name="="
            />
          </div>,
        ],
        "className": "calculator-keypad",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="function-keys"
              >
                <Button
                  clickHandler={[Function]}
                  name="AC"
                />
                <Button
                  clickHandler={[Function]}
                  name="⌫"
                />
                <Button
                  clickHandler={[Function]}
                  name="+/-"
                />
                <Button
                  clickHandler={[Function]}
                  name="√"
                />
                <Button
                  clickHandler={[Function]}
                  name="x²"
                />
                <Button
                  clickHandler={[Function]}
                  name="%"
                />
              </div>,
              <div
                className="digit-keys"
              >
                <Button
                  clickHandler={[Function]}
                  cls="key-0"
                  name="0"
                />
                <Button
                  clickHandler={[Function]}
                  name="."
                />
                <Button
                  clickHandler={[Function]}
                  name="1"
                />
                <Button
                  clickHandler={[Function]}
                  name="2"
                />
                <Button
                  clickHandler={[Function]}
                  name="3"
                />
                <Button
                  clickHandler={[Function]}
                  name="4"
                />
                <Button
                  clickHandler={[Function]}
                  name="5"
                />
                <Button
                  clickHandler={[Function]}
                  name="6"
                />
                <Button
                  clickHandler={[Function]}
                  name="7"
                />
                <Button
                  clickHandler={[Function]}
                  name="8"
                />
                <Button
                  clickHandler={[Function]}
                  name="9"
                />
              </div>,
            ],
            "className": "input-keys",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Button
                    clickHandler={[Function]}
                    name="AC"
                  />,
                  <Button
                    clickHandler={[Function]}
                    name="⌫"
                  />,
                  <Button
                    clickHandler={[Function]}
                    name="+/-"
                  />,
                  <Button
                    clickHandler={[Function]}
                    name="√"
                  />,
                  <Button
                    clickHandler={[Function]}
                    name="x²"
                  />,
                  <Button
                    clickHandler={[Function]}
                    name="%"
                  />,
                ],
                "className": "function-keys",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": "AC",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": "⌫",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": "+/-",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": "√",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": "x²",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": "%",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Button
                    clickHandler={[Function]}
                    cls="key-0"
                    name="0"
                  />,
                  <Button
                    clickHandler={[Function]}
                    name="."
                  />,
                  <Button
                    clickHandler={[Function]}
                    name="1"
                  />,
                  <Button
                    clickHandler={[Function]}
                    name="2"
                  />,
                  <Button
                    clickHandler={[Function]}
                    name="3"
                  />,
                  <Button
                    clickHandler={[Function]}
                    name="4"
                  />,
                  <Button
                    clickHandler={[Function]}
                    name="5"
                  />,
                  <Button
                    clickHandler={[Function]}
                    name="6"
                  />,
                  <Button
                    clickHandler={[Function]}
                    name="7"
                  />,
                  <Button
                    clickHandler={[Function]}
                    name="8"
                  />,
                  <Button
                    clickHandler={[Function]}
                    name="9"
                  />,
                ],
                "className": "digit-keys",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "cls": "key-0",
                    "name": "0",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": ".",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": "1",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": "2",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": "3",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": "4",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": "5",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": "6",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": "7",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": "8",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "clickHandler": [Function],
                    "name": "9",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                clickHandler={[Function]}
                name="/"
              />,
              <Button
                clickHandler={[Function]}
                name="*"
              />,
              <Button
                clickHandler={[Function]}
                name="-"
              />,
              <Button
                clickHandler={[Function]}
                name="+"
              />,
              <Button
                clickHandler={[Function]}
                cls="equalkey"
                name="="
              />,
            ],
            "className": "operator-keys",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "clickHandler": [Function],
                "name": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "clickHandler": [Function],
                "name": "*",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "clickHandler": [Function],
                "name": "-",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "clickHandler": [Function],
                "name": "+",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "clickHandler": [Function],
                "cls": "equalkey",
                "name": "=",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
